src/components/TextField/TextField.tsx

```diff
@@ -478,8 +481,8 @@ export function TextField({
     </Labelled>
   );
 
-  function handleClearButtonPress() {
-    onClearButtonClick && onClearButtonClick(id);
+  function handleClearButtonPress(event: React.MouseEvent | React.TouchEvent) {
```

Comment:
From your description it sounded like the issue is the fact the clear button is removed from the dom? Rather than exposing the event (since the issue will still occur without manually stopping propagation), we don't remove the clear button from the dom but visually hide it instead. What do you think?

Reply:
love this idea! updating the PR now

Reply:
@AndrewMusgrave We had made this change a during the AODA project but it looks like [this PR](https://github.com/Shopify/polaris-react/pull/4369) re-removed the button from the DOM because of a screen reader issue so if we go this route we should make sure to test for it. 😄 

This issue has come up in a number of places though - makes me wonder if there's a system-wide fix we could think up for `PopoverOverlay` that could handle this issue? We've had to build similarly hacky workarounds for this issue and whenever it crops up and the investigation is a tricky one for anyone without this context.

Reply:
@voiid @AndrewMusgrave I think the issue before is that this was using `VisuallyHidden` styles which has the purpose of visually hiding elements but still having them read by a screen reader

In this PR I use `visibility: hidden` which both visually hides the button AND hides it from the screen reader. I demoed it on voice over

Reply:
Oof, and it was my pr as well 😅 Looks like I did leave a [comment](https://github.com/Shopify/polaris-react/pull/4369/files#r685332423) about using `aria-hidden` if needed. So that _should_ solve the screen reader issues 🤞 

> makes me wonder if there's a system-wide fix we could think up for PopoverOverlay

Overall, yeah this is definitely the better option. The popover shouldn't be closing if the click was in it, regardless of wether is exists in the dom anymore. I'm ok with this change for a short term fix though, since I imagine a change to `PopoverOverlay` will be a lot higher effort

Reply:
Heres a demo with the rotor in voice over. as you can see the `Clear button` gets removed from the screen reader form elements when it is not visible which seems to be appropriate behaviour https://videobin.shopify.io/v/LA14z5